mock/tests/e2e/App.spec.ts

```diff
@@ -0,0 +1,424 @@
+  await expect(page.getByLabel("Login")).toBeVisible();
+});
+
+test("on page load, i dont see the input box until login", async ({ page }) => {
+  await expect(page.getByLabel("Sign Out")).not.toBeVisible();
+  await expect(page.getByLabel("Command input")).not.toBeVisible();
+  await page.getByLabel("Login").click();
+  await expect(page.getByLabel("Sign Out")).toBeVisible();
+  await expect(page.getByLabel("Command input")).toBeVisible();
+});
+test("after I type into the input box, its text changes", async ({ page }) => {
+  await page.getByLabel("Login").click();
+  await page.getByLabel("Command input").click();
+  await page.getByLabel("Command input").fill("Awesome command");
+  const mock_input = `Awesome command`;
+  await expect(page.getByLabel("Command input")).toHaveValue(mock_input);
+});
+
+test("on page load, i see a button", async ({ page }) => {
+  await page.getByLabel("Login").click();
+  await expect(
+    page.getByRole("button", { name: "Submitted 0 times" })
+  ).toBeVisible();
+});
+
+test("after I click the button, its label increments", async ({ page }) => {
+  await page.getByLabel("Login").click();
+  await expect(
+    page.getByRole("button", { name: "Submitted 0 times" })
+  ).toBeVisible();
+  await page.getByRole("button", { name: "Submitted 0 times" }).click();
+  await expect(
+    page.getByRole("button", { name: "Submitted 1 times" })
+  ).toBeVisible();
+});
+
+test("improper command input returns 'command not found' ", async ({
+  page,
+}) => {
+  await page.getByLabel("Login").click();
+  await page.getByLabel("Command input").fill("Awesome command");
+  await page.getByRole("button", { name: "Submitted 0 times" }).click();
+  const firstChild = await page.evaluate(() => {
+    const history = document.querySelector(".repl-history");
+    return history?.children[0]?.textContent;
+  });
+  expect(firstChild).toEqual("Command does not exist :(");
+});
+
+test("test that Mode command changes output", async ({ page }) => {
+  await page.getByLabel("Login").click();
+  await page.getByLabel("Command input").fill("mode");
+  await page.getByRole("button", { name: "Submitted 0 times" }).click();
+  await page.getByLabel("Command input").fill("mode");
+  await page.getByRole("button", { name: "Submitted 1 times" }).click();
+
+  const output1 = await page.evaluate(() => {
+    const history = document.querySelector(".repl-history");
+    return history?.children[0]?.textContent;
+  });
+  const output2 = await page.evaluate(() => {
+    const history = document.querySelector(".repl-history");
+    return history?.children[1]?.textContent;
+  });
+  expect(output1).toEqual("Command: modeOutput: Mode set to Verbose");
+  expect(output2).toEqual("Mode set to Brief");
+});
+
+test("test Load with improper args", async ({ page }) => {
+  await page.getByLabel("Login").click();
+  await page.getByLabel("Command input").fill("load_file");
+  await page.getByRole("button", { name: "Submitted 0 times" }).click();
```

Comment:
I deleted the submit count in my implementation so I am not sure if this will work but I am thinking maybe the {name:"Submitted 1 times"} and other things like this are not that neccessary here as there is just one button and you just need to get it and click here. Might be easier to write testing code without those